File: Introducción a la Herencia/05 - Taller 5 Python/Entregable/gestion/zoologico.py
from gestion.zona import Zona
class Zoologico:
    
    def __init__(self, nombre = "", ubicacion = ""):
        self._nombre = nombre
        self._ubicacion = ubicacion
        self._zonas = list()
    
    def cantidadTotalAnimales(self):
        cantidad1 = 0
        for zona in self._zonas:
            cantidad1 += zona.cantidadAnimales()
        return cantidad1
    
    def agregarZonas(self, zona):
        self._zonas.append(zona)
    
    def __str__(self):
        return self._nombre
    
    def getNombre(self):
        return self._nombre
    
    def setNombre(self, nombre):
        self._nombre = nombre
    
    def getUbicacion(self):
        return self._ubicacion
    
    def setUbicacion(self, ubicacion):
        self._ubicacion = ubicacion
    
    def getZona(self):
        return self._zonas
    
    def setZona(self, zonas):
        self._zonas = zonas

File: Introducción a la Herencia/04 - Taller 5 Java/Entregable/src/gestion/Zoologico.java
package gestion;

import java.util.ArrayList;
	
public class Zoologico {
	private String nombre;
	private String ubicacion;
	private ArrayList<Zona> zonas = new ArrayList<Zona>();
	
	public Zoologico(String nombre, String ubicacion) {
		this.nombre = nombre;
		this.ubicacion = ubicacion;
	}
	public Zoologico() {
		this(null, null);
	}
	
	public int cantidadTotalAnimales() {
		int cantidad1 = 0;
		for (Zona cantidad : zonas) {
			cantidad1 += cantidad.cantidadAnimales();
		}
		return cantidad1;
	}
	
	public void agregarZonas(Zona zona) {
		zonas.add(zona);
	}
	
	public String toString() {
		return nombre;
	}
	
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	
	public String getUbicacion() {
		return ubicacion;
	}
	public void setUbicacion(String ubicacion) {
		this.ubicacion = ubicacion;
	}
	
	public ArrayList<Zona> getZona(){
		return zonas;
	}
	public void setZona(ArrayList<Zona> zonas) {
		this.zonas = zonas;
	}
}
